packages/dds/map/src/directory.ts

```diff
@@ -693,17 +698,34 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         return this.localValueMaker.fromSerializable(serializable);
     }
 
+    private getParentDirectory(relativePath: string): IDirectory | undefined {
+        const absolutePath = this.makeAbsolute(relativePath);
+        const subdirs = absolutePath.substr(1).split(posix.sep);
+        if (subdirs.length === 0) {
+            return undefined;
+        }
+        subdirs.pop();
+        return this.getWorkingDirectory(subdirs.join(posix.sep));
+    }
+
+    private isSubDirectoryDeletePending(relativePath: string): boolean {
+        const parentSubDir = this.getParentDirectory(relativePath) as SubDirectory | undefined;
```

Comment:
(nit) Given that getParentDirectory() is private, I feel like it should do a cast to SubDirectory instead